### DISCUSSION TITLE
Content warnings that appear at the top of a game's page

### DISCUSSION THREAD

**Discussion by BearfaxII** [RetroAchievements/RAWeb#1087] [Ideas]
As of now, the way we handle games with potentially offensive content is by sorting them all together into the Mature hub alongside 18+ titles. While I consider this better than nothing, if only so we aren't giving zero warning whatsoever, the problems I have with this are twofold:

1. Sorting games with offensive content into that hub, particularly when it comes to the subject of especially prominent meme games like Pokemon Clover and Hong Kong 97, may convince minors to access the hub and end up exposing themselves to actual porn.
2. In some cases, it's somewhat hard to tell what exactly something is in the hub for. Someone actively looking for porn could expose themselves to a hack with nothing but 4chan humor and someone who thinks they're in for offensive memes could see naked women.

My solution to this is as follows: "content warning" tags to appear at the top of the game's page. By default, these should be on (IMO), but if someone doesn't care about what a game's content entails and wants to avoid anything they might deem a spoiler, they can hide them in their user settings (or vice versa), much like they can with the current mature content warning integrated into the site. Users at developer level may add these to specific games ala game info.

A few examples, both for the aforementioned offensive content and other subjects that may potentially be triggering:

**Content Warning:** Offensive humor (racism, islamophobia, antisemitism, nazi imagery)
`~Hack~ Pokemon Clover (Game Boy Advance)`

**Content Warning:** Sinophobia, genocide, real world photos of corpses
`~Unlicensed~ Hong Kong 97 | Hong Kong 1997 (SNES)`

**Content Warning:** Homophobic slurs
`~Hack~ EarthBound: Halloween Hack (SNES)`

**Content Warning:** Sexual assault
`177 (PC-8000/8800)`

**Content Warning:** In-universe transphobia/misgendering
`~Homebrew~ Do I Pass? (Game Boy)`

Alongside the tagging system's own benefits, it would allow the Mature hub to be reserved exclusively for porn games and retain its current functionality. Feel free to give any input.

Rough mockup of how this could look on-site:
![Example](https://user-images.githubusercontent.com/111076567/184396265-6bca333c-7135-4a61-99ab-d0db57d19ab5.png)

**Comment by suspect15** [comment]
I'm opposed to this.  I think the existing system is sufficient and appropriate.  Do people really need a warning to play any of those games?  I have to trust that if people stumble across a game with content they find objectionable, they will just stop playing.  I find this concept of warnings for everything kind of demeaning to everyone's maturity.

My default expectation is that everyone on the site is at least a teenager, so anything that could appear in say a PG-13 movie, shouldn't even be mentioned as a possible content concern.  None of the things in the above example would force a movie to be rated R, therefore shouldn't need special denotation.

If a game is just completely and inexcusably over the top, then it likely shouldn't even be allowed to have a set.  Beyond that, is this really a problem that needs to be addressed?

**Reply by BearfaxII** [reply]
Like I said, the feature would be optional; if you don't think you need warnings for anything, you can turn it off. But as a whole I think content warnings are just considerate to include. Bear in mind, we aren't simply supplying sets for games that people would have wanted to play anyway; by having games on our site, we're indirectly advertising them. There are plenty of hacks, homebrew and even full games that I don't think I ever would have played or even heard of were it not for RA. And sure, you're the type of person who would just quit something if you found its content objectionable, but there are plenty of people who can be truly negatively affected by content they would consider personally upsetting. See: PTSD.

As for the issue of games "completely and inexcusably over the top", they're another big part of the reason I'm requesting this feature. I'm inclined to agree things would be better for our site's reputation if we didn't allow them, and personally wouldn't miss them myself, but frankly I'm not sure we can convince everyone else not to allow them. Consider how popular Pokemon Clover is as a hack, to the point of it being the single most set requested hack on the site; and despite all this, it's one of, if not the most offensive game in general that currently has a page on this site. And it's definitely not the only game out there with an extreme brand of humor, either; a while back someone added a page for a Double Dragon III hack that's plot is, from my understanding, entirely founded off of transphobic jokes. I don't think we can all band together and say "let's not dev for these games", lest our site's infighting get even worse. I think the thing we're all closest to agreeing on is that content warnings would be considerate for people who aren't aware of a game's content (and imply simply having a game on the site isn't endorsing it), but not restrict the freedom of people who would want to play the game anyway.

**Reply by suspect15** [reply]
I could get on board with it if it was an option that people had to opt into.  I'm not sure how high the prevalence of people suffering negative effects from PTSD by playing certain retro games is and I question how significant a warning would even be if that were the case.

You do bring up a great point though.  I don't play hacks/unlicensed games very often, so almost all of the games I see only contain content that has been legitimately vetted and deemed appropriate for the general public by companies with an interest in it.  For games that haven't been subject to any content scrutiny I do agree that it would be reasonable to have an indication that the hack may contain some strongly objectionable themes.

I'm concerned that things might get out of control and we'd start seeing warnings for things like transphobia on Final Fantasy 7 or animal abuse on Duck Hunt.  What're you thoughts on limiting warnings to unofficially released games/hacks?

**Reply by BearfaxII** [reply]
Limiting it to unofficial releases would be reasonable, yeah. I think realistically, even if we did allow official releases to be tagged, though, examples like the ones you gave would be shut down. If we were to allow them, it should be reserved for games that contain particularly extreme content; like Custer's Revenge and the aforementioned 177, where the entire goal of the game is to rape a woman. But yes, I was primarily thinking of hacks when I made this suggestion.

**Reply by LiquifiedSnow** [reply]
Problem with going around slapping a ton of labels for warnings on games is, where does it stop? You get into the slippery slope of labelling specific things you open the door for people to start going through slapping labels for offensive content everywhere. A general warning works fine, if you want to know why it's in the hub, you could look it up and figure out why pretty quick, or you could go look at the game forums and see if anyone posted why.

**Reply by CobaltZeroni** [reply]
I don't see the problem with "slapping labels for offensive content everywhere". If obvious joke labels (or otherwise wrong labels) get made, they can be dealt with on a case-by-case basis - and I trust our developers to add content warnings responsibly.

**Reply by LiquifiedSnow** [reply]
Might as well start labeling Mario games with animal abuse and mega man with elderly abuse if were gonna mark up everything with warnings. Any game with swords and any sort of combat gonna have to slap a violence warning on.

**Reply by CobaltZeroni** [reply]
I think most people would agree that Mario and Mega Man are too cartoony to necessitate warnings, but having them for realistic depictions of gore or the like seems very reasonable. Content warnings don't need to be applied to literally everything, just stuff that's common triggers.

**Reply by LiquifiedSnow** [reply]
Saying "too cartoony to necessitate warnings" is the same as me saying "it's just a video game it has a warning it doesnt need anymore. Or it has ratings already we don't need our own"

**Reply by televandalist** [reply]
![custer](https://user-images.githubusercontent.com/45054151/184522626-65e11e5a-a99c-451c-81b4-12ee541719e0.png)
Too cartoony?

**Reply by CobaltZeroni** [reply]
Honestly, adding ESRB content blurbs would be an easy way to quickly populate content warnings.
Re: Custer, that's one case where the actual subject matter would warrant a warning no matter how cartoony it was.

**Reply by suspect15** [reply]
> Honestly, adding ESRB content blurbs would be an easy way to quickly populate content warnings. Re: Custer, that's one case where the actual subject matter would warrant a warning no matter how cartoony it was.

Does anyone actually pay attention to ESRB content warnings?  I always assumed those were intended to caution parents buying games for their kids.

**Comment by Jamiras** [comment]
see also #802

**Comment by Xeinok-Pals** [comment]
Is there a data source we could automate pulling in the ESRB/etc. content warning categories from?

**Reply by suspect15** [reply]
I don't think the issue is really with games that have ESRB ratings, so I don't think that addresses the concern.  Although I'm squarely on the side of "no" on this particular discussion, it seems like the real concerns are the homebrew, hacks and unlicensed games.

As Snow said, the problem with things like this are that they inevitably get taken past what probably makes sense - see Super Mario Bros 2 in the LGBT Representation Hub because of Birdo (the purple dinosaur that spits eggs at you and who has a bow on its head).  While they are often well intentioned to begin, they tend to get off track and not really ever addressed after that.

I see this as an ultimately low-value at best thing that won't be uniformly and consistently implemented and end up being more of a loose end than anything else.

**Comment by Jamiras** [comment]
Partially implemented for Mature Content (#2952) and Epilepsy Warning (#3990). These warnings appear in the sidebare below the boxart.
